src/assets/TokenListController.ts

```diff
@@ -317,19 +251,28 @@ export class TokenListController extends BaseController<
           (token) => !duplicateSymbols.includes(token.symbol),
         );
         for (const token of uniqueTokenList) {
-          tokenList[token.address] = token;
+          const formattedToken: RawToken = {
+            ...token,
+            aggregators: formatAggregatorNames(
+              (token.aggregators as AggregatorKey[]) || [],
+            ),
+            iconUrl: formatIconUrlWithProxy({
+              chainId: this.chainId,
+              tokenAddress: token.address,
+            }),
+          };
+          tokenList[token.address] = formattedToken;
```

Comment:
Since `tokenList` can comes from state, this would directly modify state, which could cause issues. What do you think about making a copy of it when you pull it out of state or use `reduce` to build a modified version of the token list?

Reply:
Good catch. I applied the spread operator to create copies for `tokenList`